File: dataset/human/python/file_9150.py
from collections import deque
S=input()
N=len(S)
i=0
flag=1
ans=[]
buf=""
while i<N:
    if (i+9<N)and((S[i]=="d")and(S[i+1]=="r")and(S[i+2]=="e")and(S[i+3]=="a")and(S[i+4]=="m")and(S[i+5]=="e")and(S[i+6]=="r")and(S[i+7]=="a")and(S[i+8]=="s")and(S[i+9]=="e")):
        if i+10==N:
            break
        elif (S[i+10]=="d")or(S[i+10]=="e"):
            i=i+10
        elif (S[i+10]=="r"):
            i=i+5
        else:
            flag=0
            break
    elif (i+6<N)and((S[i]=="d")and(S[i+1]=="r")and(S[i+2]=="e")and(S[i+3]=="a")and(S[i+4]=="m")and(S[i+5]=="e")and(S[i+6]=="r")):
        if i+7==N:
            break
        elif (S[i+7]=="d")or(S[i+7]=="e"):
            i=i+7
        else:
            flag=0
            break
    elif (i+4<N)and((S[i]=="d")and(S[i+1]=="r")and(S[i+2]=="e")and(S[i+3]=="a")and(S[i+4]=="m")):
        if i+5==N:
            break
        elif (S[i+5]=="d"):
            i=i+5
        else:
            flag=0
            break
    elif (i+5<N)and((S[i]=="e")and(S[i+1]=="r")and(S[i+2]=="a")and(S[i+3]=="s")and(S[i+4]=="e")and(S[i+5]=="r")):
        if i+6==N:
            break
        elif (S[i+6]=="e")or(S[i+6]=="d"):
            i=i+6
        else:
            flag=0
            break
    elif (i+4<N)and((S[i]=="e")and(S[i+1]=="r")and(S[i+2]=="a")and(S[i+3]=="s")and(S[i+4]=="e")):
        if i+5==N:
            break
        elif (S[i+5]=="e")or(S[i+5]=="d"):
            i=i+5
        else:
            flag=0
            break
    else:
        flag=0
        break
    #print(i)
    #input()
if flag==1:
    print("YES")
else:
    print("NO")

File: dataset/human/cpp/file_9150.cpp
#include<bits/stdc++.h>
using namespace std;
int main(){
    int n,m;
    cin>>n>>m;
    char a[n+1][n+1];
    char b[m+1][m+1];
    getchar();
    for(int i=0;i<n;i++){
        for(int j=0;j<n;j++)
        cin>>a[i][j];
    }
     for(int i=0;i<m;i++){
        for(int j=0;j<m;j++)
        cin>>b[i][j];
    }
     /* for(int i=0;i<n;i++){
        for(int j=0;j<n;j++)
        cout<<a[i][j];
        cout<<endl;
    }
     for(int i=0;i<m;i++){
        for(int j=0;j<m;j++)
        cout<<b[i][j];
            cout<<endl;
    }*/
    int flag=0;
    for(int i=0;i<n-m+1;i++){
    for(int j=0;j<n-m+1;j++){
        flag=0;
        for(int s=0,k=i;s<m;s++,k++){
            for(int t=0,d=j;t<m;t++,d++){
                if(b[s][t]!=a[k][d]){
                    flag=1;
                    break;
                }
            }
            if(flag==1){
                break;
            }
        }
        if(flag==0){
            cout<<"Yes"<<endl;
            return 0;
        }
    }
    }
    cout<<"No"<<endl;
}
